### ISSUE TITLE
OpenSSL 1.1.1 build Alpha OpenVMS V8.3 - $GETTIM_PREC()

### ISSUE THREAD

**Issue by mgdaniel** [openssl/openssl#7229]
A build difference between OpenVMS 8.4 and V8.3

```
A missing symbol during link.

8< snip 8<
WRITE OPT_FILE "[.crypto]threads_win.OBJ,-"
WRITE OPT_FILE "[.crypto]uid.OBJ"
CLOSE OPT_FILE
LINK /MAP='F$PARSE(".MAP","ossl$libcrypto0101_shr.EXE")'
\>/NODEBUG/NOTRACEBACK/SHARE=ossl$libcrypto0101_shr.EXE
libcrypto.opt-translated/OPT,                ossl$libcrypto-components.OPT/OPT
%LINK-W-NUDFSYMS, 1 undefined symbol:
%LINK-I-UDFSYM, 	sys$gettim_prec
%LINK-W-USEUNDEF, undefined symbol sys$gettim_prec referenced
	in psect $LINK$ offset %X00000090
	in module rand_vms file
WASD_ROOT:[src.openssl-1_1_1.crypto.rand]rand_vms.OBJ;1
%MMS-F-ABORT, For target ossl$libcrypto0101_shr.EXE, CLI returned abort
status: %X10648268.
%MMS-F-ABORT, For target all, CLI returned abort status: %X10EE8034.
```

-----------------------------------------------------------------------------------------------

My solution clipped from .openssl-1_1_1.crypto.rand]rand_vms.c

```
/* MGD 15-SEP-2018

   $GETTIM_PREC is not available until V8.4

%LINK-W-NUDFSYMS, 1 undefined symbol:
%LINK-I-UDFSYM,    sys$gettim_prec
%LINK-W-USEUNDEF, undefined symbol sys$gettim_prec referenced
   in psect $LINK$ offset %X00000090
   in module rand_vms file
WASD_ROOT:[src.openssl-1_1_1.crypto.rand]rand_vms.OBJ;1

*/
#if __CRTL_VER >= 80400000
    sys$gettim_prec(&data.time);
#else
    sys$gettim((void*)&data.time);
#endif
    return rand_pool_add(pool, (unsigned char *)&data, sizeof(data), 0);
}

int rand_pool_add_additional_data(RAND_POOL *pool)
{
    struct {
        CRYPTO_THREAD_ID tid;
        uint64_t time;
    } data = { 0 };

    /*
     * Add some noise from the thread id and a high resolution timer.
     * The thread id adds a little randomness if the drbg is accessed
     * concurrently (which is the case for the <master> drbg).
     */
    data.tid = CRYPTO_THREAD_get_current_id();
#if __CRTL_VER >= 80400000
    sys$gettim_prec(&data.time);
#else
    sys$gettim((void*)&data.time);
#endif
```

-----------------------------------------------------------------------------------------------

3) I'm also thinking that the additional entropy provided by the
$GETTIM_PREC() over $GETTIM() is probably negligible, in which case dropping
the $GETTIM_PREC() entirely and using only $GETTIM() might improve
portability of compiled code between V8.n version of VMS.

Just my AU$0.05 worth (we don't have $0.02 coins here).

~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

**Comment by levitte** [comment]
Thanks, will fix.

Re the `$GETTIM_PREC` vs `$GETTIM` though, you're right that it might be bordering silly, but am also fairly certain that there are those who will keep pointing out that there is a "high precision" (utterly depending on your hardware) timer, so I think I'll keep the possibility.

**Comment by levitte** [comment]
Fix is in #7230, feel free to try it out (it also contains other fixes that I know affect OpenVMS as well)
